projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp

```diff
@@ -0,0 +1,430 @@
+		Trap();
+	}
+
+	/* Start measurement */
+	eSensorResult = cn0398.start();
+
+	if(eSensorResult != SENSOR_ERROR_NONE) {
+		PRINT_SENSOR_ERROR(DEBUG_MESSAGE, eSensorResult);
+		Trap();
+	}
+
+	/* Publish AD7124 ADC data using MQTT */
+	MQTTPublishCN0398(cn0398);
+
+	return 0;
+}
+
+/*!
+ * @brief      Application Callback
+ *
+ * @details    Called by the framework layer (adi_wifi_noos.c) when an event occurs.
+ *
+ * @param [in] pCBParam : Callback parameter (unused)
+ *
+ * @param [in] Event : Event of type #ADI_WIFI_AT_CMDCODE.
+ *
+ * @param [in] pArg : Callback argument (unused)
+ *
+ */
+void adi_wifi_AplicationCallback(void * pCBParam,
+                                 uint32_t nEvent,
+                                 void * pArg)
+{
+	switch (nEvent) {
+	case CMD_NONE:
+		/* Read the publish Data. */
+		adi_wifi_ParseSubscriberData();
+		adi_wifi_GetData((uint8_t*)ui8S, &packet_size);
+		rec_flag = true;
+		break;
+	case CMD_HW_ERROR:
+		break;
+	default:
+		break;
+	}
+
+	return;
+}
+
+static void MQTTPublishCN0398(CN0398 pCn)
+{
+	ADI_WIFI_RESULT eResult;
+	uint32_t        nPacketId = 0u;
+	uint8_t         nConnectionFailFlag;
+
+	/* Initialize the ESP8266 Wi-Fi module and establish a TCP connection */
+	InitWiFiConnection();
+
+	/* Send MQTT "CONNECT" message to broker */
+	eResult = adi_wifi_radio_MQTTConnect(&gMQTTConnect);
+	DEBUG_RESULT("Error connecting to the MQTT broker.\r\n",
+	             eResult,
+	             ADI_WIFI_SUCCESS);
+
+	/* Subscribe to receive topic */
+	eResult = adi_wifi_radio_MQTTSubscribe(&gSubscribeConfig);
+	DEBUG_RESULT("Error subscribing to hidden_channel topic.\r\n",
+	             eResult,
+	             ADI_WIFI_SUCCESS);
+
+	/* Calibration sequence */
+	sprintf((char *)ui8S, "Initialization complete!\r\n\n");
+	gPublishConfig.pMQTTData = (uint8_t*)ui8S;
+	gPublishConfig.nMQTTDataSize = strlen((char *)ui8S);
+	adi_wifi_radio_MQTTPublish(&gPublishConfig);
+
+	sprintf((char *)ui8S, "Do you want to perform pH calibration [y/N]?\n\n");
+	gPublishConfig.pMQTTData = (uint8_t*)ui8S;
+	gPublishConfig.nMQTTDataSize = strlen((char *)ui8S);
+	adi_wifi_radio_MQTTPublish(&gPublishConfig);
+	while(!rec_flag) {
+		adi_wifi_DispatchEvents(1000);
+	}
+	rec_flag = false;
+
+	if(payload_ptr[0] == 'y' || payload_ptr[0] == 'Y')
+		pCn.calibrate_ph();
+	else {
+		pCn.use_nernst = true;
+		sprintf((char *)ui8S, "Do you want to load default calibration?[y/N].");
+		gPublishConfig.pMQTTData = (uint8_t*)ui8S;
+		gPublishConfig.nMQTTDataSize = strlen((char *)ui8S);
+		adi_wifi_radio_MQTTPublish(&gPublishConfig);
+		sprintf((char *)ui8S, "If not[N], the Nernst equation will be used.\n");
+		gPublishConfig.pMQTTData = (uint8_t*)ui8S;
+		gPublishConfig.nMQTTDataSize = strlen((char *)ui8S);
+		adi_wifi_radio_MQTTPublish(&gPublishConfig);
+		while(!rec_flag)
+			adi_wifi_DispatchEvents(1000);
+		rec_flag = false;
+		if(payload_ptr[0] == 'y' || payload_ptr[0] == 'Y')
+			pCn.use_nernst = false;
+	}
```

Comment:
remember the if/else special case for brackets:
```
if (condition) {
        do_this();
        do_that();
} else {
        otherwise();
}
```
https://www.kernel.org/doc/html/v4.10/process/coding-style.html#placing-braces-and-spaces